Remove default shared model config directory in Pose reader

lochhh opened the pull request:
Omitting the `model_root` argument will now make the reader search only for local model config directories.

* Remove redundant argument from legacy pose test
* Ensure single error path for missing model config
* Update pre-commit ruff version

Closes #20

lochhh committed:
Require `model_root` arg in Pose reader

glopesdev reviewed (commented):
@lochhh This looks good to me, can you confirm whether the tests are still failing? Unfortunately I cannot access the logs anymore as they have expired.

lochhh committed:
Update pre-commit ruff version

lochhh committed:
Add required `model_root` in test schema initialisations

glopesdev commented:
@lochhh Thank you for the tests, I added https://github.com/SainsburyWellcomeCentre/aeon_api/pull/38 as an alternative implementation where `model_root` is disabled by default.

I realize this is a breaking change from the previous behaviour, but feels like a natural one. Otherwise, for the last test the actual path is irrelevant, so even a simple `.` would be enough to pass the test since only local config directories are used.

glopesdev committed:
Make `model_root` optional in Pose reader (#38)

* Make `model_root` optional in Pose reader

Omitting the argument will make the reader search only for local model
config directories.

* Remove redundant argument from legacy pose test

* Ensure single error path for missing model config

glopesdev reviewed (commented):
@lochhh LGTM thanks, I guess the only thing I would do now is change the title of the PR, given that strictly we didn't really make the argument required in the end.

I'm thinking perhaps we can just reword the description from the last PR?

glopesdev committed:
Avoid using Optional type annotation

lochhh commented:
Yup feel free to edit the PR title and merge. Thanks again @glopesdev !

glopesdev reviewed (approved):
Thanks, I noticed there is a new untested code path due to the branch in error handling, but probably best to leave it for https://github.com/SainsburyWellcomeCentre/aeon_api/pull/26 so not to disturb the test restructuring done there.

lochhh commented:
> Thanks, I noticed there is a new untested code path due to the branch in error handling, but probably best to leave it for #26 so not to disturb the test restructuring done there.

This test case has been added in 88116cfaf32de0cc309deacc8539be54d29b4e06
